fix: fixed tests

pablocael opened the pull request:
Cpp tests were failing for the right reason (I have added exception for non initialized indices and did not handle it in tests before).

Commit:
fix: fixed tests

Dobatymo commented:
Can you use `EXPECT_THROW`?
or maybe call fail when no exception is thrown?
also I guess the second tree2 is redundant and will never be reached.

Commit:
fix: using expect throw and no throw in cpp tests

pablocael commented:
> Can you use `EXPECT_THROW`? or maybe call fail when no exception is thrown? also I guess the second tree2 is redundant and will never be reached.

Good idea. I have simplified and added another no throw test as well to make sure it will not throw if we set data in index.
Fixed on 14ac4d6e11b6f570df6815c5c978f42432214eb2

Thanks
